Lib/test/test_free_threading/test_set.py

```diff
@@ -0,0 +1,43 @@
+import unittest
+
+from threading import Thread, Barrier
+from unittest import TestCase
+
+from test.support import threading_helper
+
+
+@threading_helper.requires_working_threading()
+class TestSet(TestCase):
+    def test_repr_clear(self):
+        """Test repr() of a set while another thread is calling clear()"""
+        NUM_ITERS = 10
+        NUM_REPR_THREADS = 10
+        barrier = Barrier(NUM_REPR_THREADS + 1)
+        s = {1, 2, 3, 4, 5, 6, 7, 8}
+        set_reprs = []
```

Comment:
This may be too smart, so feel free to reject, but... (1/4)

```suggestion
        expected = {repr(set()), repr(s)}
```